APIReviewFoxy.md

```diff
@@ -0,0 +1,601 @@
+*   [https://github.com/ros2/common_interfaces/blob/master/std_msgs/msg/UInt64.msg](https://github.com/ros2/common_interfaces/blob/master/std_msgs/msg/UInt64.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/std_msgs/msg/UInt64MultiArray.msg](https://github.com/ros2/common_interfaces/blob/master/std_msgs/msg/UInt64MultiArray.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/std_msgs/msg/UInt8.msg](https://github.com/ros2/common_interfaces/blob/master/std_msgs/msg/UInt8.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/std_msgs/msg/UInt8MultiArray.msg](https://github.com/ros2/common_interfaces/blob/master/std_msgs/msg/UInt8MultiArray.msg)
+
+
+#### std_srvs
+
+Srv
+
+
+
+*   [https://github.com/ros2/common_interfaces/blob/master/std_srvs/srv/Empty.srv](https://github.com/ros2/common_interfaces/blob/master/std_srvs/srv/Empty.srv)
+*   [https://github.com/ros2/common_interfaces/blob/master/std_srvs/srv/SetBool.srv](https://github.com/ros2/common_interfaces/blob/master/std_srvs/srv/SetBool.srv)
+*   [https://github.com/ros2/common_interfaces/blob/master/std_srvs/srv/Trigger.srv](https://github.com/ros2/common_interfaces/blob/master/std_srvs/srv/Trigger.srv)
+
+
+#### stereo_msgs
+
+Comment: In addition to DisparityImage.msg , I always wished stereo_msgs would have a message type for stereo image pairs. Instead of a stereo camera driver having to publish two separate image topics, then having every subscriber needing to use an message synchronizer to capture the matching image pairs via matching timestamp, the driver could publish both images over a stereo image pair message. There could be common fixed length size message for an common array of two images, or perhaps in a separate message package, a message type for collecting a dynamic number of Image.msg for multi camera rigs, like motion capture or trinary cameras for multi view reconstruction.
+
+**Suggested Action:** None. The design was chosen to send the parallel streams of data to avoid publishing data multiple times or requiring extra subscriptions. The simplest use case is if you want to process one of the two stereo images. If they’re published as a pair, you will have to receive both and then discard the entire second image which wastes a lot of resources. On the other hand we can simply have tools that collect the parallel streams of messages and give you callbacks when you get the groupings you want without causing extra bandwidth.
+
+Msg
+
+
+
+*   [https://github.com/ros2/common_interfaces/blob/master/stereo_msgs/msg/DisparityImage.msg](https://github.com/ros2/common_interfaces/blob/master/stereo_msgs/msg/DisparityImage.msg)
+    *   Comment:  Header that might be removed in a future release, maybe this one?
+    *   **Suggested Action:** Review removal of header.
+
+
+#### trajectory_msgs
```

Comment:
This package contains only messages related to joints. So would `joint_trajectory_msgs` be a better name? A typical trajectory is also a path (see nav_msgs), so this might avoid some naming confusion.